proto/src/determined/api/v1/api.proto

```diff
@@ -509,6 +509,78 @@ service Determined {
       tags: "Internal"
     };
   }
+  // Get the absolute training length until the searcher needs the trial to
+  // report a validation.
+  rpc GetTrialSearcherTrainUntil(GetTrialSearcherTrainUntilRequest)
+      returns (GetTrialSearcherTrainUntilResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/trials/{trial_id}/searcher/train_until"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+  // Reports to the searcher that the trial has completed the current
+  // requested amount of training with the given searcher validation
+  // metric.
+  rpc ReportTrialSearcherValidation(ReportTrialSearcherValidationRequest)
+      returns (ReportTrialSearcherValidationResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trials/{trial_id}/searcher/report_validation"
+      body: "searcher_metric"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+  // Reports to the searcher that the trial has completed the current
+  // requested amount of training with the given searcher validation
+  // metric.
+  rpc ReportTrialSearcherEarlyExit(ReportTrialSearcherEarlyExitRequest)
+      returns (ReportTrialSearcherEarlyExitResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trials/{trial_id}/searcher/early_exit"
+      body: "early_exit"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+  // For bookkeeping, updates the progress towards to current requested searcher
+  // training length.
+  rpc ReportTrialProgress(ReportTrialProgressRequest)
+      returns (ReportTrialProgressResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trials/{trial_id}/searcher/report_progress"
```

Comment:
Progress is the nature of a trial not a searcher.
```suggestion
      post: "/api/v1/trials/{trial_id}/progress"
```

Reply:
This progress can also be used to be seen by users although we don't support it, right?

Reply:
> If it's early stopped, it's early stopped in the middle of the whole planned precedure.

yeah i suppose you could report progress as a percent of max_length and if it is early stopped it just never reaches 100%? is this what you mean.

Reply:
exactly what I mean.